### ISSUE TITLE
Bar Chart visualization settings show an error and do not allow changes to be made

### ISSUE THREAD

**Issue by jp-swinetech** [metabase/metabase#33936]
### Describe the bug

When creating a new bar chart with a single value, the Visualization Settings do not work correctly. The settings show an error icon and do not allow any changes to be made.

I believe this may have been introduced in a [recent PR](https://github.com/metabase/metabase/pull/32875) -- This PR changed the `getDefaultColumns` function in frontend/src/metabase/visualizations/lib/settings/graph.js, which is present in the javascript stack trace included below.

To work around the issue, I was able to switch to a "Row" visualization type, change the "Y-axis" label to "Count", then switch back to a "Bar" visualization. After doing that, the Bar visualization settings worked as expected. I was only able to do this if I converted the question to a SQL question.

### To Reproduce

- Create a new Question using the Sample Database "Accounts" table. Add "Count of rows" as a summary metric and click visualize.
- Change the Visualization type to "Bar"
- Attempt to access the Visualization Settings. Observe an error icon and no settings available to change.

### Expected behavior

Ideally, with a single row and column, I would expect the "Bar" visualization to show a single bar with a "Count" label. Barring that (no pun intended), I expect the visualization settings to allow me to change settings. 

### Logs

Javascript console stack trace:
```
react-dom.production.min.js:209 TypeError: Cannot read properties of undefined (reading 'name')
    at M (utils.js:154:42)
    at MG (graph.js:45:7)
    at Object.getProps (graph.js:657:20)
    at settings.js:163:22
    at settings.js:186:7
    at Array.map (<anonymous>)
    at ct (settings.js:185:6)
    at S (visualization.js:116:10)
    at n.value (ChartSettings.jsx:164:14)
    at n.value (ChartSettings.jsx:285:26)
```

### Information about your Metabase installation

```
Browser: Chrome 116.0.5845.188
Operating System: Windows
Databases: Observed on both H2 Sample DB and with a BigQuery backend
Metabase Version: metabase/metabase-enterprise:v1.47.1 docker image
Metabase Hosting Environment: Docker
Metabase Internal Database: Observed on both H2 and Postgres
```

### Severity

Annoying

**Comment by nemanjaglumac** [comment]
If this helps, `v0.46.6` still doesn't show the single bar, but it at leasts shows the settings sidebar
![image](https://github.com/metabase/metabase/assets/31325167/53a781bd-daa6-4028-9c05-9576496ab148)

While 47 doesn't render the sidebar because of an error. It says "Something's gone wrong" instead.

**Comment by npfitz** [comment]
The issue regarding the error will be handled in https://github.com/metabase/metabase/issues/34762, but the functionality of Bar charts supporting metric only queries will have it's own issue in  https://github.com/metabase/metabase/issues/34762

**Comment by npfitz** [comment]
This was included in 47.6, but setting to 47.7 so that it is included in release notes
